draft-ietf-lake-edhoc.md

```diff
@@ -2139,6 +2139,72 @@ While this key update method provides forward secrecy it does not give as strong
+
+~~~~~~~~~~~~~~~~~~~~~~~
+    +-------------------> START
+    |                       |
+    |                       |
+    |    Receive invalid    | Send message_1
+    |    or error msg       |
+    |    (send error msg)   v
+ABORTED <---------------- WAIT_M2
+    ^                       |
+    |                       | Receive message_2
+    |    (send error msg)   v
+    +-------------------- RCVD_M2
+    ^                       |
+    |                       |
+    |    Receive invalid    | Send message_3
+    |    or error msg       |
+    |    (send error msg)   v
+    +------------------- COMPLETED
+                            |
+                            | (Receive message_4)
+                            v
+                         PERSISTED <-- Successful decryption and
+                                       verification of an incoming
+                                       application message protected
+                                       with derived application keys
+~~~~~~~~~~~~~~~~~~~~~~~
+{: artwork-align="center"}
+
+## Responder State Machine
+
+The responder transitions from START to RCVD_M1 upon the reception of a message_1. If a processing error occurred on message_1, the Responder sends back an error message. This includes sending error message with error code 2 (Wrong Selected Cipher Suite) if the received message_1 was generated with an unsupported cipher suite. Once in RCVD_M1, the Responder processes and prepares message_2 for sending. It transitions to WAIT_M3 by successfully sending message_2. The Responder then processes the incoming message, and if an error occurs during the processing sends an error message back and discontinues the protocol. If an error message is received instead, the protocol is discontinued but the error message sending is skipped. Successful processing of message_3 transitions the state machine into COMPLETED. If an application profile mandates message_4, the Responder sends it and transitions into PERSISTED. Otherwise, it directly transitions into PERSISTED state.
+
+~~~~~~~~~~~~~~~~~~~~~~~
+                          START
+                            |
+                            | Receive message_1
+                            |
+         (send error msg)   v
+ABORTED <---------------- RCVD_M1
+    ^                       |
+    |    Receive invalid    |
+    |    or error msg       | Send message_2
+    |    (send error msg)   v
+    +-------------------- WAIT_M3
+    ^                       |
+    |                       |
+    |                       | Receive message_3
+    |                       |
+    |    (send error msg)   v
+    +------------------- COMPLETED
+                            |
+                            | (Send message_4)
+                            v
+                         PERSISTED
+~~~~~~~~~~~~~~~~~~~~~~~
+{: artwork-align="center"}
+
```

Comment:
Please see below proposed updated text, consistent with the updated figure further below.

---

The responder transitions from START to RCVD_M1 upon the reception of a valid message_1.

If any error is encountered while processing message_1, the Responder sends back an error message and discontinues the protocol by transitioning from RCVD_M1 to ABORTED. The sent error message has error code 2 (Wrong Selected Cipher Suite) in case the selected cipher suite is not supported by the Responder, or if the Responder supports a cipher suite more preferred by the Initiator than the selected cipher suite.

Otherwise, the Responder prepares message_2 for sending. If any error is encountered while preparing message_2, the Responder sends an error message and discontinues the protocol by transitioning from RCVD_M1 to ABORTED. Once message_2 is successfully prepared, the Responder sends it by transitioning from RCVD_M1 to WAIT_M3.

In case the next incoming message is invalid or is an error message, the Responder transitions from WAIT_M3 to ABORTED. In the former case, the Responder sends an error message. In either case, the Responder discontinues the protocol.

Upon receiving a valid message_3, the Responder transitions from WAIT_M3 to RCVD_M3 and processes the message. 

If any error is encountered while processing message_3, the Responder sends back an error message and discontinues the protocol by transitioning from RCVD_M3 to ABORTED.

Otherwise, in case of successful processing, the Responder transitions to COMPLETED.

Once in COMPLETED, the Responder transitions either to PERSISTED or to PREP_M4 for preparing message_4 (if supported by the application profile). Once in PREP_M4, if any error is encountered while preparing message_4, the Responder sends an error message and discontinues the protocol by transitioning to ABORTED. Otherwise, the Responder sends message_4 by transitioning from PREP_M4 to PERSISTED.

```
~~~~~~~~~~~~~~~~~~~~~~~
                          START
                            |
                            | Receive message_1
         Processing error   |
         (send error msg)   v
ABORTED <---------------- RCVD_M1
    ^                       |
    |    Receive invalid    | Send message_2
    |    or error msg       |
    |    (send error msg)   v
    +-------------------- WAIT_M3
    ^                       |
    |                       | Receive message_3
    |    Processing error   |
    |    (send error msg)   v
    +-------------------- RCVD_M3
    ^                       |
    |                       |
    |                       v
    |                     COMPLETED -----------------+
    |                       |                        |
    |    Processing error   |                        |
    |    (send error msg)   v                        |
    +-------------------- (PREP_M4)                  |
                            |                        |
                            | (Send message_4)       |
                            v                        |
                          PERSISTED <----------------+
~~~~~~~~~~~~~~~~~~~~~~~
```